src/main/java/edu/escuelaing/ieti/matchmaking/persistence/UserRepository.java

```diff
@@ -1,18 +1,19 @@
 package edu.escuelaing.ieti.matchmaking.persistence;
 
+import edu.escuelaing.ieti.matchmaking.exception.MatchmakingException;
 import edu.escuelaing.ieti.matchmaking.model.User;
 
 import java.util.List;
 
 public interface UserRepository {
 
-    User create(User user);
+    User create(User user) throws MatchmakingException;
```

Comment:
Does not make sense to have only one type of exception for throwing every possible error. Consider separating you exceptions in two main types: Business and System. Also, determine if they will be checked or unchecked